### PR TITLE
More 1.17-pre1 stuff

### PR DESCRIPTION
**Pull request by apple502j**
Shouldn't conflict with #2500

### CODE REVIEW

File: mappings/net/minecraft/screen/ScreenHandler.mapping [hunk 1]
```diff
@@ -10,6 +10,7 @@ CLASS net/minecraft/class_1703 net/minecraft/screen/ScreenHandler
 	FIELD field_30730 EMPTY_SPACE_SLOT_INDEX I
 		COMMENT A special slot index value ({@value}) indicating that the player has clicked outside the main panel
 		COMMENT of a screen. Used for dropping the cursor stack.
+	FIELD field_34024 nonce I
```
Comment (modmuss50) [inline mappings/net/minecraft/screen/ScreenHandler.mapping]: I’m sure this doesn’t mean what comes to my mind (from the uk). Something else might better here as I don’t know what this word is supposed to mean.

**Review by modmuss50 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/screen/ScreenHandler.mapping (hunk 1 above)
Comment (apple502j) [reply mappings/net/minecraft/screen/ScreenHandler.mapping]: It's an integer that is used to validate if something is in sync. Better name maybe? (Implemented as incremental id, but can be any integer as long as it changes per packet)

File: mappings/net/minecraft/util/UserCache.mapping [hunk 2]
```diff
@@ -8,6 +8,8 @@ CLASS net/minecraft/class_3312 net/minecraft/util/UserCache
 	FIELD field_25724 accessCount Ljava/util/concurrent/atomic/AtomicLong;
 	FIELD field_25805 LOGGER Lorg/apache/logging/log4j/Logger;
 	FIELD field_29788 MAX_SAVED_ENTRIES I
+	FIELD field_33860 pendingRequests Ljava/util/Map;
+	FIELD field_33861 executor Ljava/util/concurrent/Executor;
```
Comment (liach) [inline mappings/net/minecraft/util/UserCache.mapping:12]: Is `executor` good enough, or should we name it like `gameEngine` etc?

File: mappings/net/minecraft/screen/ScreenHandler.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/screen/ScreenHandler.mapping]: Is this a value to ensure modifications are up-to-date? Think about java collections' `modCount`.

File: mappings/net/minecraft/client/network/ClientPlayerEntity.mapping [hunk 3]
```diff
@@ -98,3 +98,4 @@ CLASS net/minecraft/class_746 net/minecraft/client/network/ClientPlayerEntity
 	METHOD method_3151 getMountJumpStrength ()F
 	METHOD method_33689 init ()V
 	METHOD method_7290 dropSelectedItem (Z)Z
+		ARG 1 dropAll
```
Comment (liach) [inline mappings/net/minecraft/client/network/ClientPlayerEntity.mapping]: I like the `entireStack` from #2500 better.

File: mappings/net/minecraft/server/network/ServerPlayerEntity.mapping [hunk 4]
```diff
@@ -197,6 +197,8 @@ CLASS net/minecraft/class_3222 net/minecraft/server/network/ServerPlayerEntity
 		ARG 1 player
 	METHOD method_34225 onSpawn ()V
 	METHOD method_34879 areClientChatColorsEnabled ()Z
+	METHOD method_37413 dropSelectedItem (Z)Z
+		ARG 1 dropAll
```
Comment (liach) [inline mappings/net/minecraft/server/network/ServerPlayerEntity.mapping]: Compare to (`drop`)`entireStack` from #2500

File: mappings/net/minecraft/network/PacketByteBuf.mapping [hunk 5]
```diff
@@ -683,3 +683,8 @@ CLASS net/minecraft/class_2540 net/minecraft/network/PacketByteBuf
 		COMMENT
 		COMMENT @return the read chunk position
 		COMMENT @see #writeChunkPos(ChunkPos)
+	METHOD method_37435 writeOptional (Ljava/util/Optional;Ljava/util/function/BiConsumer;)V
+		ARG 1 value
+		ARG 2 serializer
+	METHOD method_37436 readOptional (Ljava/util/function/Function;)Ljava/util/Optional;
```
Comment (liach) [inline mappings/net/minecraft/network/PacketByteBuf.mapping:697]: Mind add javadocs for this method and add the two methods to the javadoc's table of this class?

File: mappings/net/minecraft/block/entity/SkullBlockEntity.mapping [hunk 6]
```diff
@@ -5,6 +5,7 @@ CLASS net/minecraft/class_2631 net/minecraft/block/entity/SkullBlockEntity
 	FIELD field_12088 sessionService Lcom/mojang/authlib/minecraft/MinecraftSessionService;
 	FIELD field_12089 userCache Lnet/minecraft/class_3312;
 	FIELD field_31363 SKULL_OWNER_KEY Ljava/lang/String;
+	FIELD field_34029 executor Ljava/util/concurrent/Executor;
```
Comment (liach) [inline mappings/net/minecraft/block/entity/SkullBlockEntity.mapping:8]: Same game engine vs executor question

File: mappings/net/minecraft/screen/ScreenHandler.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/screen/ScreenHandler.mapping]: `version` might be better; we used `version` back when vanilla had a multithreading-error-prone tag container implementation that used an int field to check if a tag reference is up-to-date.

Comment (Juuxel) [reply mappings/net/minecraft/screen/ScreenHandler.mapping]: Even `revision` could work (IMO sounds more automated than `version` if you think about software dev)

File: mappings/net/minecraft/screen/ScreenHandler.mapping [hunk 7]
```diff
@@ -85,8 +85,8 @@ CLASS net/minecraft/class_1703 net/minecraft/screen/ScreenHandler
 	METHOD method_37418 getSlotIndex (Lnet/minecraft/class_1263;I)Ljava/util/OptionalInt;
 		ARG 1 inventory
 		ARG 2 index
-	METHOD method_37421 getNonce ()I
-	METHOD method_37422 incrementAndGetNonce ()I
+	METHOD method_37421 getRevision ()I
+	METHOD method_37422 incrementAndGetRevision ()I
```
Comment (liach) [inline mappings/net/minecraft/screen/ScreenHandler.mapping]:
```suggestion
	METHOD method_37422 nextRevision ()I
```

**Review by liach — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

File: mappings/net/minecraft/client/realms/gui/screen/RealmsSubscriptionInfoScreen.mapping [hunk 8]
```diff
@@ -17,6 +17,7 @@ CLASS net/minecraft/class_4425 net/minecraft/client/realms/gui/screen/RealmsSubs
 	FIELD field_26521 DAY_TEXT Lnet/minecraft/class_2561;
 	FIELD field_26522 DAYS_TEXT Lnet/minecraft/class_2561;
 	FIELD field_32127 EXTEND_JAVA_REALMS_URL Ljava/lang/String;
+	FIELD field_34033 UNKOWN_TEXT Lnet/minecraft/class_2561;
```
Comment (liach) [inline mappings/net/minecraft/client/realms/gui/screen/RealmsSubscriptionInfoScreen.mapping]:
```suggestion
	FIELD field_34033 UNKNOWN_TEXT Lnet/minecraft/class_2561;
```

File: mappings/net/minecraft/world/gen/HeightContext.mapping [hunk 9]
```diff
@@ -1,3 +1,8 @@
 CLASS net/minecraft/class_5868 net/minecraft/world/gen/HeightContext
+	FIELD field_34030 minY I
+	FIELD field_34031 maxY I
```
Comment (enbrain) [inline mappings/net/minecraft/world/gen/HeightContext.mapping]: Isn't this height?

File: mappings/net/minecraft/client/realms/gui/screen/RealmsSubscriptionInfoScreen.mapping (hunk 8 above)
Comment (apple502j) [reply mappings/net/minecraft/client/realms/gui/screen/RealmsSubscriptionInfoScreen.mapping]: This is from #2500 but it's fine :P

File: mappings/net/minecraft/world/gen/HeightContext.mapping (hunk 9 above)
Comment (apple502j) [reply mappings/net/minecraft/world/gen/HeightContext.mapping]: Lemme see, if this is the case getters need renaming

Comment (apple502j) [reply mappings/net/minecraft/world/gen/HeightContext.mapping]: Yeah, seems like the getter needs to be fixed.

Comment (liach) [reply mappings/net/minecraft/world/gen/HeightContext.mapping]: I recall we mostly use `height` as an alternative to `deltaY`; it is a relative than an abstract value

File: mappings/net/minecraft/client/realms/gui/screen/RealmsSubscriptionInfoScreen.mapping (hunk 8 above)
Comment (IotaBread) [reply mappings/net/minecraft/client/realms/gui/screen/RealmsSubscriptionInfoScreen.mapping]: Oh, didn't realize it. Sorry

File: mappings/net/minecraft/world/gen/HeightContext.mapping (hunk 9 above)
Comment (apple502j) [reply mappings/net/minecraft/world/gen/HeightContext.mapping]: Okay, confirmed previous mapping is wrong and this is a relative value (YOffset uses it in BelowTop as `method_30458() + getMinY() - 1 - offset`

Comment (liach) [reply mappings/net/minecraft/world/gen/HeightContext.mapping]: Ah thank you guys for correcting :heart:

**Review by enbrain — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**